tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -0,0 +1,255 @@
+    my ($self) = @_;
+    my $ip_out = script_output('ip route show | grep -Eo "src\s+([0-9.]*)\s+" | head -1 | cut -d\' \' -f 2', 30);
+    set_var('SERVER_IP', $ip_out);
+    bmwqemu::save_vars();
+
+    # Wait for destination to be ready
+    $self->set_mutex_lock('DST_READY_TO_START');
+
+    my $agent_ip = $self->get_var_from_parent("AGENT_IP");
+
+    rke2_server_setup($self, $agent_ip);
+    deploy_kubevirt_manifests();
+    run_virt_tests();
+    generate_test_report();
+    upload_results();
+}
+
+# Mutex lock/unlock functionality
+sub set_mutex_lock {
+    my ($self, $lock_signal) = @_;
+
+    mutex_lock($lock_signal);
+    mutex_unlock($lock_signal);
+}
+
+sub rke2_server_setup {
+    my ($self, $agent_ip) = @_;
+
+    systemctl('stop apparmor.service');
+    # systemctl('stop SuSEfirewall2.service');
+    assert_script_run('ssh-keygen -b 2048 -t rsa -q -N "" -f ~/.ssh/id_rsa') unless (-e "~/.ssh/id_rsa");
+    exec_and_insert_password("ssh-copy-id -o StrictHostKeyChecking=no -i /root/.ssh/id_rsa.pub root\@$agent_ip");
+
+    # Ensure SUSE certificates are installed on the node
+    ensure_ca_certificates_suse_installed();
+
+    # Install downstream kubevirt packages
+    zypper_call('in kubernetes1.18-client') if (script_run('rpmquery kubernetes1.18-client'));
+
+    # RKE2 deployment on server node
+    my $rke_ver = get_var('INSTALL_RKE2_VERSION');
+    if ($rke_ver) {
+        assert_script_run("curl -sfL https://get.rke2.io | INSTALL_RKE2_VERSION=$rke_ver sh -");
+    } else {
+        assert_script_run('curl -sfL https://get.rke2.io | sh -');
+    }
+    systemctl('enable rke2-server.service');
+    systemctl('start rke2-server.service');
+
+    # Start rke2-server service ready
+    mutex_create('RKE2_SERVER_START_READY');
+
+    # Wait for rke2-agent service to be ready
+    $self->set_mutex_lock('RKE2_AGENT_START_READY');
+
+    assert_script_run('cp /etc/rancher/rke2/rke2.yaml ~/.kube/config');
+    assert_script_run('kubectl get nodes');
+
+    # Create registries ready
+    script_run("cat > /etc/rancher/rke2/registries.yaml <<__END
+mirrors:
+  registry.suse.de:
+    endpoint:
+      - http://registry.suse.de
+configs:
+  registry.suse.de:
+    tls:
+      insecure_skip_verify: true
+__END");
+
+    assert_script_run("scp /etc/rancher/rke2/registries.yaml root\@$agent_ip:/etc/rancher/rke2/registries.yaml");	
+    mutex_create('CREATE_REGISTRIES_READY');
+
+    # Restart RKE2 service and check the service is active well after restart
+    systemctl('restart rke2-server.service');
+    check_service_status();
+}
+
+sub check_service_status {
+    # Check RKE2 services & kubevirt test environment
+    systemctl('is-active rke2-server.service');
+}
+
+sub deploy_kubevirt_manifests {
+    my $os_version = script_output("cat /etc/os-release | grep VERSION= | awk -F\" '{print \$2}'");
+    zypper_call("ar http://download.suse.de/ibs/Devel:/Virt:/SLE-" . $os_version . "/SUSE_SLE-" . $os_version . "_Update_standard/ devel");
```

Comment:
Regarding from which repo to get the manifests packages, I am asking whether we need to support multiple ways:
- in official testing, take 15sp4 for example, we need to install packages from [15sp4](https://download.suse.de/ibs/SUSE:/SLE-15-SP4:/GA/standard)
- in devel testing, we use pkg in devel repo
Please keep tuned.

Reply:
@nanzhg  I have talked with dev and RM, the final solution is:
- in official testing, we should use official product repo, eg 15sp4 daily build or 15sp3 MU(what the host is installed ) , the maniests package will be in container module while virt-tests pkg will be in development-tools
- in devel testing, we use pkg in devel repo

Reply:
Thank you for following up this issue. Has switched to official repo by default for OSD test. 
`zypper_call("ar https://download.suse.de/ibs/SUSE:/SLE-" . $os_version . ":/GA/standard/ " . $repo_alias);`

Reply:
To avoid misunderstanding, let me clarify it again explicitly. We are likely to need to support 3 different kinds of kubevirt testing, development product test(15sp4 now), MU product test and devel test.
- development product test: for 15sp4 official product OSD testing, the repo to install kubevirt from should be the existing reposotories on 15sp4 host after scc registration and update_pkg.pm , that is the tested 15sp4 target build +15sp4 update channel , but it is NOT 15SP4 GA repo
- MU product test(code support can be added after the PR, but should consider how it can be easily done with limited code change): eg 15sp3 MU, we should install kubevirt from 15sp3 official MU channel+incidents' update repo(can be configured from testsuite, just as current MU incident test does)
- devel test(code support can be added after the PR): the repo to install kubevirt pkg should be Devel:/Virt: for relevant product

Reply:
OK, we have implemented to get manifest packages for development product test. For MU product test, the full repo URL can be tuned by dynamic getting OS version. If the repo url is totally different, we can introduce a new setting to distinguish the different product test.